app.py

```diff
@@ -0,0 +1,137 @@
+        'format': 'json',
+        'type': 'flat',
+        'csvDelimiter': '',
+        'fields[0]': 'netid',
+        'fields[1]': 'record_id',
+        'fields[2]': 'eligibility_screening_complete',
+        'fields[3]': 'consent_form_complete',
+        'fields[4]': 'enrollment_questionnaire_complete',
+        'filterLogic': filter_logic,
+        'rawOrLabel': 'raw',
+        'rawOrLabelHeaders': 'raw',
+        'exportCheckboxLabel': 'false',
+        'exportSurveyFields': 'false',
+        'exportDataAccessGroups': 'false',
+        'returnFormat': 'json'
+    }
+    response = requests.post(REDCAP_API_URL, data=data)
+    response.raise_for_status()
+
+    if len(response.json()) == 0:
+        return None
+
+    assert len(response.json()) == 1, \
+        f"Error: Multiple records matching {filter_logic}"
+
+    return response.json()[0]
+
+def register_net_id(net_id: str) -> str:
+    """
+    Returns the REDCap record ID of the participant newly registered under the
+    given *net_id*.
+    """
+    # REDCap enforces that we must provide a non-empty record ID. Because we're
+    # using `forceAutoNumber` in the POST request, we do not need to provide a
+    # real record ID.
+    values = [{'netid': net_id, 'record_id': 'record ID cannot be blank'}]
+    data = {
+        'token': REDCAP_API_TOKEN,
+        'content': 'record',
+        'format': 'json',
+        'type': 'flat',
+        'overwriteBehavior': 'normal',
+        'forceAutoNumber': 'true',
+        'data': json.dumps(values),
+        'returnContent': 'ids',
+        'returnFormat': 'json'
+    }
+    response = requests.post(REDCAP_API_URL, data=data)
+    response.raise_for_status()
+    return response.json()[0]
+
+
+def generate_survey_link(record_id: str) -> str:
+    """
+    Returns a generated survey link to the eligibility screening instrument for
+    the given *record_id*.
+    """
+    data = {
+        'token': REDCAP_API_TOKEN,
+        'content': 'surveyLink',
+        'format': 'json',
+        'instrument': 'eligibility_screening',
+        'event': 'enrollment_arm_1',  # TODO subject to change
+        'record': record_id,
+        'returnFormat': 'json'
+    }
+    response = requests.post(REDCAP_API_URL, data=data)
+    response.raise_for_status()
+    return response.text
+
+
+@app.route('/')
+def main():
+    # Get NetID from Shibboleth data
+    net_id = request.remote_user
+
+    while not net_id:
+        # TODO: Redirect to Shibboleth login
+        net_id = 'KaasenG'
+        pass
+
+    try:
+        user_data = fetch_user_data(net_id)
+
+    except Exception as e:
+        app.logger.warning(f'Failed to fetch REDCap data for user {net_id}: {e}')
+
+        return """
+            Error: Something went wrong. Please contact Husky Coronavirus Testing support by
+            emailing <a href=\"mailto:huskytest@uw.edu\">huskytest@uw.edu</a> or by calling
+            <a href=\"tel:+12066162414\">(206) 616-2414</a>.
+            """
```

Comment:
They're harmless, but FYI you don't need to escape double quotes inside a triple-quoted string.

Reply:
I originally had single quoted strings here :)